Assets/Scripts/Utility/MacroHelper.cs

```diff
@@ -217,6 +217,23 @@ public static class MacroHelper
 
         #endregion
 
+        #region fields (some macros need state (e.g. %fn2 depends on %fn1)
```

Comment:
Macros that need state or context should be implemented using a macro context provider (MCP) class. Having this class state in here is less than ideal, @Nystul-the-Magician is this just temporary? Commit comment says "using static field to manage interaction for now" so that's my guess. When you get to it I will help out if you find my tutorial guide is not good enough. Just let me know.

Reply:
no need helped - already implemented MCP class for other stuff. This is a placeholder - since we need to find out how vanilla implements these macros I thought this would be ok to have it this way for now just to have something. As soon as someone finds out the correct behaviour we can implement this cleanly